src/utils/uncapitalize.ts

```diff
@@ -2,13 +2,14 @@ import { charAt } from '../native/char-at.js'
 import { join } from '../native/join.js'
 import { slice } from '../native/slice.js'
 import { toLowerCase } from '../native/to-lower-case.js'
+import type { UncapitalizeSTS } from '../internal/intrinsic.js'
 
 /**
  * Uncapitalizes the first letter of a string. This is a runtime counterpart of `Uncapitalize<T>` from `src/types.d.ts`.
  * @param str the string to uncapitalize.
  * @returns the uncapitalized string.
  * @example uncapitalize('Hello world') // 'hello world'
  */
-export function uncapitalize<T extends string>(str: T): Uncapitalize<T> {
+export function uncapitalize<T extends string>(str: T): UncapitalizeSTS<T> {
```

Comment:
@p9f any opinions here? Maybe we should leave the changes in these functions to another PR so we can discuss.
If we export these types we must think through, if we ship with different types we are gonna be breaking a concept of matching type and runtime.
I'm thinking if there's a way to make our type utilities behave more like those of TS and leave them unchanged?

Reply:
>  if we ship with different types we are gonna be breaking a concept of matching type and runtime.

i am not sure to understand this part :/ afaiu this whole PR is making the type more precise for non literal string - but where do we have an inconsistency between runtime and types?

Reply:
I don't think our types will work with `string` types (eg: `CamelCase<string>`) without fixing the bad behavior that TS built-in functions (eg: `Uppercase`) have. 

Any utility that uses a built-in func will have a return type like `Uppercase<string>` instead of `string`. This is weird. Also, we pipe types in a lot of places, so it will cause issues where the outputs/inputs don't align between piped types

Reply:
I understand it now.. I'll think through it.
Still not convinced the value is worth the added LOCs and complexity 😅

Reply:
IMO it is a critical mistake to not handle `string` type in a lib called `string-ts` -- and the intrinsic/built-in string utilities cannot do that.